backend/controller/dal/testdata/go/fsmnext/fsmnext_test.go

```diff
@@ -0,0 +1,36 @@
+package fsmnext
+
+import (
+	"testing"
+
+	"github.com/TBD54566975/ftl/go-runtime/ftl"
+	"github.com/TBD54566975/ftl/go-runtime/ftl/ftltest"
+	"github.com/alecthomas/assert/v2"
+)
+
+func TestProgression(t *testing.T) {
```

Comment:
do you see any value in adding invalid progression cases? e.g. {B, C, C}, {B, A}

Reply:
Added 👍